src/iceberg_functions/iceberg_table_properties_functions.cpp

```diff
@@ -0,0 +1,295 @@
+#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
+#include "duckdb/catalog/catalog_entry/table_function_catalog_entry.hpp"
+#include "duckdb/parser/tableref/joinref.hpp"
+#include "duckdb/parser/tableref/table_function_ref.hpp"
+#include "duckdb/parser/expression/constant_expression.hpp"
+#include "duckdb/planner/expression/bound_reference_expression.hpp"
+#include "duckdb/planner/operator/logical_get.hpp"
+#include "duckdb/planner/operator/logical_comparison_join.hpp"
+#include "duckdb/common/file_system.hpp"
+#include "iceberg_metadata.hpp"
+#include "iceberg_functions.hpp"
+#include "iceberg_utils.hpp"
+#include "../include/storage/irc_table_entry.hpp"
+#include "duckdb/main/client_context.hpp"
+#include "storage/irc_catalog.hpp"
+#include "storage/irc_table_entry.hpp"
+#include "storage/iceberg_table_information.hpp"
+#include "storage/iceberg_transaction_data.hpp"
+#include "storage/irc_table_entry.hpp"
+#include "storage/irc_transaction.hpp"
+#include "storage/iceberg_table_information.hpp"
+#include "metadata/iceberg_table_metadata.hpp"
+
+#include <string>
+
+namespace duckdb {
+
+struct SetIcebergTablePropertiesBindData : public TableFunctionData {
+	optional_ptr<ICTableEntry> iceberg_table;
+	case_insensitive_map_t<string> properties;
+	vector<string> remove_properties;
+};
+
+struct SetIcebergTablePropertiesGlobalTableFunctionState : public GlobalTableFunctionState {
+public:
+	SetIcebergTablePropertiesGlobalTableFunctionState() {};
+
+	static unique_ptr<GlobalTableFunctionState> Init(ClientContext &context, TableFunctionInitInput &input) {
+		return make_uniq<SetIcebergTablePropertiesGlobalTableFunctionState>();
+	}
+
+	idx_t property_count = 0;
+	// FIXME: this is probably super dumb, but since properties are just string->string, we will keep them in
+	// memory like this for now.
+	// need to keep the properties ordered when returning them in case there are 2048+
+	// that way we don't duplicate certain properties
+	vector<pair<string, string>> all_properties;
+	bool all_properties_initialized = false;
+	bool properties_set = false;
+	bool properties_removed = false;
+};
+
+static optional_ptr<ICTableEntry> GetTableEntry(ClientContext &context, string &input_string) {
+
+	auto qualified_name = QualifiedName::ParseComponents(input_string);
+	optional_ptr<ICTableEntry> iceberg_table = nullptr;
+	auto default_db = DatabaseManager::GetDefaultDatabase(context);
```

Comment:
I don't think this is what we want to do, we can check inside GetIcebergTableEntry, see my earlier comment